File: app/models/qdm/clinical_impression.rb
module QDM
  # app/models/qdm/clinical_impression.rb
  class ClinicalImpression < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :status
    field :statusReason
    field :code
    field :description
    field :subject
    field :encounter
    field :effective
    field :date
    field :assessor
    field :previous
    field :problem
    field :investigation
    field :protocol
    field :summary
    field :finding
    field :prognosisQDM::CodeableConcept
    field :prognosisReference
    field :supportingInfo
    field :note
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/ClinicalImpression.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const ClinicalImpressionSchema = DataElementSchema({
  identifier: Any,
  status: Any,
  statusReason: Any,
  code: Any,
  description: Any,
  subject: Any,
  encounter: Any,
  effective: Any,
  date: Any,
  assessor: Any,
  previous: Any,
  problem: Any,
  investigation: Any,
  protocol: Any,
  summary: Any,
  finding: Any,
  prognosisCodeableConcept: Any,
  prognosisReference: Any,
  supportingInfo: Any,
  note: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'ClinicalImpression' },

});

module.exports.ClinicalImpressionSchema = ClinicalImpressionSchema;
module.exports.ClinicalImpression = mongoose.model('ClinicalImpression', ClinicalImpressionSchema);
